### PR TITLE
Add Battle.net OAuth backend

### PR DESCRIPTION
**Pull request by cburmeister**
This commit adds an OAuth backend for Battle.net's OAuth APIs.

https://develop.battle.net/documentation/api-reference/oauth-api

### CODE REVIEW

File: loginpass/battlenet.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+"""
+    loginpass.battlenet
+    ~~~~~~~~~~~~~~~~
+
+    Loginpass Backend for Battle.net.
+
```
Comment (lepture) [inline loginpass/battlenet.py]: Could you add the document links here.

Comment (cburmeister) [reply loginpass/battlenet.py]: Yeah!

**Comment by cburmeister** [comment]
@lepture let me know if there's anything else I can do here

**Comment by lepture** [comment]
Thanks

**Pull request merged**